atqa-framework/tests/pages/BasePage.ts

```diff
@@ -0,0 +1,89 @@
+import { Locator } from '../../interfaces/Locator.interface';
+
+export interface IterableConstant {
+  [key: string]: string;
+}
+
+const { I } = inject();
+
+export class BasePage {
+  BROWSER_URLS = {
+    POPUP: '/popup.html',
+    ACCOUNTS: '/accounts.html',
+    CHROME_SYSTEM: 'chrome://system/',
+    CHROMIUM_EXTENSIONS: 'chrome://extensions/',
+    FIREFOX_EXTENSIONS: '#/runtime/this-about:debuggingfirefox',
+  };
+
+  BROWSER_SELECTORS = {
+    CHROMIUM_EXTENSION_VALUE_BUTTON: { id: 'extensions-value-btn' },
+    CHROMIUM_EXTENSIONS_LIST: { xpath: '(//div[@class="stat-value"])[5]' },
+    CHROMIUM: {
+      UPDATE_EXTENSION_BUTTON: {
+        shadow: ['extensions-manager', 'extensions-toolbar', '#updateNow'],
+      },
+      DETAILS_BUTTON: {
+        shadow: [
+          'extensions-manager',
+          'extensions-item-list',
+          'extensions-item',
+          '#detailsButton',
+        ],
+      },
+      DEV_MODE_TOGGLE: {
+        shadow: ['extensions-manager', 'extensions-toolbar', '#devMode'],
+      },
+    },
+    EDGE: {
+      SERVICE_WORKER_RESTART_BUTTON: {
+        css: '[aria-label="Turn on Keeper Wallet"]',
+      },
+      DETAILS_BUTTON: { xpath: '//span[contains(text(),"Details")]' },
+      DEV_MODE_TOGGLE: { css: '#developer-mode' },
+      EXTENSION_MENU_BUTTON: {
+        xpath: '(//button[@aria-label= "Extensions menu"])[2]',
+      },
+      UPDATE_EXTENSION_BUTTON: { xpath: '//span[contains(text(),"Reload")]' },
+    },
+    FIREFOX: {
+      MANIFEST_EMAIL_TEXT: 'support@wavesplatform.com',
+    },
+  };
+
+  AVAILABILITY = {
+    AVAILABLE: 'available',
+    NOT_AVAILABLE: 'not available',
+  };
+
+  // TODO: Fix any type
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  getItemFromLocalStorage = async (keyName: string): Promise<any> => {
```

Comment:
If you set only `string` values in `localStorage` with `setItemToLocalStorage`, you can assume that the return type `getItemToLocalStorage` will also be `string`.

Reply:
_executeScript_ is kinda hack of codeceptjs for getting access to browser api and it always returns void, so I can't throw string type in this way

Reply:
I do not understand, what is the problem to return the correct type from the function?

Reply:
It is possible to use this construction if you are sure about the return type
```javascript
return I.executeScript((key) => {
      return window.localStorage.getItem(key);
    }, `data-qa:${keyName}`) as unknown as string;
```

Reply:
It's `string | null` actually. Yes, it's useful if it's defined